### DISCUSSION TITLE
Sqlite: Allow registering an impl for a no-arg sql function

### DISCUSSION THREAD

**Discussion by brassy-endomorph** [diesel-rs/diesel#3179] [Ideas]
It may be that the docs are missing something clear, but I want to be able to create a no-arg sql function and register a rust function to execute it. For example, this should work:

```rust
no_arg_sql_function!(uuidv4, Binary);

fn foo(conn: &SqliteConnection) -> Result<(), _> {
   uuidv4::register_impl(&conn, || Vec::from(Uuid::new_v4().as_bytes())
}
```

It is possible to register impls for functions with arguments but not without.

**Comment by weiznich** [comment]
This is not supported using the 1.4.x release. The master branch/2.0.0-rc.0 release allows this with a slightly different syntax:

```rust
diesel::sql_function!(fn uuidv4() -> Binary);

fn foo(conn: &SqliteConnection) -> Result<(), _> {
   uuidv4::register_impl(&conn, || Vec::from(Uuid::new_v4().as_bytes())
}

```
